programs/mango-v4/src/instructions/perp_place_order.rs

```diff
@@ -109,8 +110,31 @@ pub fn perp_place_order(ctx: Context<PerpPlaceOrder>, order: Order, limit: u8) -
 
     let now_ts: u64 = Clock::get()?.unix_timestamp.try_into().unwrap();
 
-    // TODO apply reduce_only flag to compute final base_lots, also process event queue
-    require!(order.reduce_only == false, MangoError::SomeError);
+    let pp = account.perp_position(perp_market_index)?;
+    let mut effective_pos = pp.effective_base_position_lots();
+    effective_pos = if effective_pos >= 0 {
+        effective_pos - pp.asks_base_lots
```

Comment:
Add a comment with the reasoning for including open orders in one direction and event queue lots. 

Shouldn't the casing depend on `side`? Consider trying to place a bid when you have 10 base_lots and 15 lots in open asks. Right now the effective_pos would be -5, meaning we could add 5 more bids? That would be incorrect.

Also, I'd prefer a .max(0) here and a .min(0) below. 

Finally, maybe this should be called the base_lots_limit and only be computed in the `if reduce_only` case?